staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -288,12 +288,15 @@ func (s preparedGenericAPIServer) Run(stopCh <-chan struct{}) error {
 	// Wait for all requests to finish, which are bounded by the RequestTimeout variable.
 	s.HandlerChainWaitGroup.Wait()
 
+	// wait for server listener to be closed
+	<-serverDoneCh
```

Comment:
I think we could read from this chan before `s.HandlerChainWaitGroup.Wait()` so that we know that the lister has been closed.